registry/client/src/settings/List.js

```diff
@@ -0,0 +1,88 @@
+import React, {Children, cloneElement} from 'react';
+import {useMediaQuery, makeStyles} from '@material-ui/core';
+import {
+    List,
+    Datagrid,
+    EditButton,
+    BooleanField,
+    SimpleList,
+    SelectField,
+    TextField,
+} from 'react-admin';
+
+import {types} from './dataTransform';
+
+const ListActionsToolbar = ({children, ...props}) => {
+    const classes = makeStyles({
+        toolbar: {
+            alignItems: 'center',
+            display: 'flex',
+            marginTop: -1,
+            marginBottom: -1,
+        },
+    });
+
+    return (
+        <div className={classes.toolbar}>
+            {Children.map(children, button => cloneElement(button, props))}
+        </div>
+    );
+};
+
+const Pagination = () => (<div />);
+
+const SourceValueField = (props) => {
+    const secret = '********';
+
+    if (props.record.secret) {
+        return secret;
+    }
+
+    switch (props.record.meta.type) {
+        case types.choices: {
+            return (<SelectField {...props} choices={props.record.meta.choices} />);
+        }
+        case types.boolean: {
+            return (<BooleanField {...props} />);
+        }
+        case types.password: {
+            return secret;
+        }
+        default: {
+            return (<TextField {...props} />);
+        }
+    }
+}
+
+const PostList = props => {
+    const isSmall = useMediaQuery(theme => theme.breakpoints.down('sm'));
+    return (
+        <List
+            {...props}
+            exporter={false}
+            bulkActionButtons={false}
+            pagination={<Pagination />}
+        >
+            {isSmall
+                ? (
+                    <SimpleList
+                        primaryText={record => record.key}
+                        secondaryText={record => record.value}
+                    />
+                ) : (
+                    <Datagrid rowClick="edit" optimized>
```

Comment:
Could you attach a screenshot of this page?

Reply:
<img width="1440" alt="Screenshot 2020-10-20 at 12 48 11" src="https://user-images.githubusercontent.com/57893569/96570085-978d8b00-12d2-11eb-8af5-97ce0b067c2d.png">

Reply:
can we create one field (object) for `auth.openId`